python_modules/dagster/dagster/_core/storage/db_io_manager.py

```diff
@@ -174,28 +173,39 @@ def _get_table_slice(
         if context.has_asset_key:
             asset_key_path = context.asset_key.path
             table = asset_key_path[-1]
-            if output_context_metadata.get("schema") and self._schema:
-                raise DagsterInvalidDefinitionError(
-                    f"Schema {output_context_metadata.get('schema')} "
-                    "specified via output metadata, but conflicting schema "
-                    f"{self._schema} was provided via run_config. "
-                    "Schema can only be specified one way."
-                )
-            elif output_context_metadata.get("schema"):
+            # schema order of precedence: metadata, key_prefix, I/O manager 'schema' config
+            if output_context_metadata.get("schema"):
+                if self._schema:
+                    context.log.warn(
```

Comment:
these warnings might be excessive - i'm happy to remove them

Reply:
could also replace with a debug log saying what the schema is

Reply:
I recommend removing them. I'd reserve warnings for situations where we or the user are doing something inadvisable, and I think having more specific values override the defaults is perfectly advisable.

Reply:
ok!